### PR TITLE
enhance es exporter service

### PR DESCRIPTION
**Pull request by pgvishnuram**
## Description

ES Exporter is set to run as root it's not needed anymore , current change will run the service a less privileged user.

**changes**
* run as non root service to removed unnecessary root usage

## Related Issues
Yet to define

## Testing

Yet to define

## Merging

Yet to define

### CODE REVIEW

**Comment by pgvishnuram** [comment]
Need to write test cases for the same self @pgvishnuram

File: charts/elasticsearch/templates/exporter/es-exporter-deployment.yaml [hunk 1]
```diff
@@ -50,7 +50,8 @@ spec:
 {{ toYaml (default .Values.global.platformNodePool.tolerations .Values.tolerations) | indent 8 }}
       restartPolicy: {{ .Values.exporter.restartPolicy }}
       securityContext:
-        runAsUser: 0
+        runAsNonRoot: true
+        runAsUser: {{ .Values.securityContext.runAsUser }}
```
[the remark this answers is not in the document]
Comment (pgvishnuram) [reply charts/elasticsearch/templates/exporter/es-exporter-deployment.yaml:54]: if you have open policy agent or any equivalent policy enforcement - it wont allow to run as root.
In openshift we have scc privileges to run as anyuid - that allows users to set and run with custom id's.

**Review by santoshr1016 — APPROVED** [review]
So you are running as userid 1000, which is safe.. 👍 
Test case would complete it..

File: tests/chart_tests/test_elasticsearch.py [hunk 2]
```diff
@@ -205,3 +205,34 @@ def test_nginx_es_client_network_selector_with_logging_sidecar_enabled(
...
+            ],
+        )
+
+        assert len(docs) == 1
+        doc = docs[0]
+        es_index = doc["data"]["nginx.conf"]
+        assert doc["kind"] == "ConfigMap"
+        assert "vector.$remote_user.*/$1" in es_index
+
+    def test_elasticsearch_exporter_securitycontext_defaults(self, kube_version):
```
Comment (danielhoherd) [inline tests/chart_tests/test_elasticsearch.py:225]: Can we get one more test here that uses a custom `.Values.securityContext.runAsUser`?

Comment (danielhoherd) [inline tests/chart_tests/test_elasticsearch.py:225]: Can we get one more test here that uses a custom `.Values.securityContext.runAsUser`?

Comment (pgvishnuram) [reply tests/chart_tests/test_elasticsearch.py:225]: yes sure

Comment (pgvishnuram) [reply tests/chart_tests/test_elasticsearch.py:225]: yes sure

**Review by danielhoherd — APPROVED** [review]

**Pull request merged**